Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -422,3 +566,33 @@ def ReadFile(path):
         LOG_ACTION.log(LogType.Error, "Exception opening file " + path + " Error Code: " + str(error.errno) + " Error: " + error.message + error.strerror)
     return content
 
+def ConvertToUtf8(item):
+    return str(item).encode('utf-8')
+
+def GetWorkspaceListFromStateFile():
+    stateFileFullPath = os.path.join(PATH_NPM_STATE, FNAME_NPM_STATE)
+    fileContents = ReadFile(stateFileFullPath)
+    if fileContents is None:
+        return []
+    wsList = filter(IsUUID, fileContents.splitlines())
+    return map(ConvertToUtf8, wsList)
+
+def IsUUID(uuidStr):
+    try:
+        uuidOut = re.match('[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}\Z', uuidStr, re.I)
```

Comment:
Can you send me a link for the documentation you're looking at which provides these format constraints? If possible, we should use [python's UUID class](http://stackoverflow.com/questions/534839/how-to-create-a-guid-uuid-in-python) to check this (i.e. try casting to a UUID and if it throws an exception, it is not a valid UUID).